addons/hr_timesheet/models/project.py

```diff
@@ -141,7 +141,7 @@ def _compute_effective_hours(self):
     def _compute_progress_hours(self):
         for task in self:
             if (task.planned_hours > 0.0):
-                task_total_hours = task.effective_hours + task.subtask_effective_hours
+                task_total_hours = sum(task.timesheet_ids.mapped('unit_amount')) + task.subtask_effective_hours
```

Comment:
As you've said, computing the `subtask_effective_hours` is a recursive operation, and it's using the task's `effective_hours` + `subtask_effective_hours`. So `subtask_effective_hours` inherently contains the rounding of the `effective_hours`. So even if we don't see verbatim the rounding in the compute of the `subtask_effective_hours`, it's implicit in the compute of the `child_task.effective_hours`. To illustrate my point, you can try out:
- Set the `planned_hours` on a task to 11min.
- Create a subtask, and on that subtask create a timesheet for 11min.
- Go back to the parent task and observe that we still have the same rounding error. (98% progress instead of 100%).

So the solution to your problem inherently comes from the fact that `planned_hours` is never really rounded anywhere. There multiple alternative solutions:
- You can round it during the saving (2 lines in the write of Task in project.py in project module)
```py
if vals.get('planned_hours'):
    vals['planned_hours'] = round(vals['planned_hours'], 2)
```
This saves in the DB the planned rounded.
- You can round it in `_compute_progress_hours`, when assigning to the `task.progress`
```py
task.progress = round(100.0 * task_total_hours / round(task.planned_hours, 2), 2)
```
This just impact the progress bar. The value in the DB is saved unrounded.

I would tend to prefer the modification in the write, as it would fix other problematic behaviours. For example:
```py
task.overtime = max(task_total_hours - task.planned_hours, 0)
```
happens to work even with the rounding errors, because `task_total_hours - task.planned_hours` is negative in our context...

Reply:
Side notes:
Update the commit and PR msg with the correct minutes to reproduce the error. :) 3 min is converted to 0.05 in float, which is not a problematic value.

Reply:
Hi @hamzaislam101,
After discussion with @kitan191, we came to the conclusion that there is no *real* solution, that covers all cases, so feel free to implement either solution (either removing the round on the `effective_hours` as you initially did in your first PR, or round the `planning_hours` as I've suggested) and open the PR. @xavierbol will give you his input.

Reply:
Force push on this one, so Xavier has the whole context. And the other PR is in 15, not 14. :)